### ISSUE TITLE
TLC for the index

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1567]
tl;dr the index has a lot of data in it, and we need to invest some time to make sure it's nice an responsive to commands

###

We need to schedule some time to do index maintenance. Probably one day, and ideally first thing on the list, if we can, e.g. over this weekend if I have time or on Monday, ahead of other things on the issue list, apart from any small things that can be quickly done.

As you know, we have been taking a closer look at things over the last week or so, and after some warnings the other day of things being down I was taking a closer look at the index response times. Then I was also doing more work with the index recently when loading the wikidata stuff into it.

There has been no problem with the index stability or query speed, and no noticeable effect on downtime, which is good (and our plans for more API monitoring would help to tell us if there were any, in future).

However, I found today that I could not issue some commands to the index and get them to execute fast enough, so they failed. This turns out to be because we now have too much different types of data for the machine we are running on. We had already considered turning on another one again, but this is not immediately necessary - the index just needs some maintenance.

The problem is, some of our logs and caches have become so large that the machine can't process a command on them before it times out! So the data is fine, things keep working, but we have to do this maintenance now to make sure we can keep it running smooth.

Today I had to shut it all down to get it to respond to other commands, hence the few minutes of downtime on live, and about ten minutes on dev.

it has been about two years since we did any cleaning up of these things as well. At first I thought maybe the wikidata stuff was too big, but once processed it actually came in smaller than a few of our OAB indexes. But when I tried to remove one of the old OAB caches, that is when the command timeouts started. Nobody would have seen query timeouts at that point - this is internal backend stuff I am talking about timeouts on. However they would have seen public timeouts when I had to bring the index down, as that affects all machines that connect to it.

We have almost 200 million log records! And a few hundred GB of cached data! Pretty good going, really. But a lot of it is old and can be got rid of now. And once we get the test/log/alerting stuff configured, I will set up some auto-pruning of those as well.

So this fits reasonably well into the work we have scheduled now anyway - backend sorting, alert setup, etc. But will make it all take a little longer. As mentioned above, hopefully only a day longer.

I may have to cause some more temporary downtime on live at some point to get it finished. But this time I will know in advance, so will put a "down for 5 mins" warning on the front page of the site if I have to do it.

**Comment by markmacgillivray** [comment]
I closed this but re-opening to discuss - index work is done, but also we should schedule turning on another index machine. Can discuss that then create a new issue if suitable, then close this one.
